### PR TITLE
Adding Greenhouse Window, Climate Control,  Camouflage Spray Can and Improve Recipes

### PR DESCRIPTION
**Pull request by Nedelosk**

### CODE REVIEW

**Comment by Nedelosk** [comment]
@mezz  Is anything wrong with this PR ?

File: src/main/java/forestry/api/climate/IClimateSourceProvider.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+package forestry.api.climate;
+
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public interface IClimateSourceProvider {
+	
+	IClimateSource getClimateSource();
+	
+	World getWorld();
```
Comment (mezz) [inline src/main/java/forestry/api/climate/IClimateSourceProvider.java]: This name will cause issues.  You should have this interface extend `ILocatable`.

**Comment by mezz** [comment]
I only see one thing, I made a comment.
This is more than I can review reliably. I can pull this if you are available to quickly fix any crashes that come up.

**Comment by Nedelosk** [comment]
@mezz I have pulled this and I am available to quickly fix any crashes that come up.

**Pull request merged**